dev_docs/contributing/kibana_http_api_guidelines.mdx

```diff
@@ -0,0 +1,502 @@
+
+These guidelines will help you build APIs that are easy to use, whether someone's automating with scripts or managing infrastructure at scale.
+
+## Terraform Provider Developer-Friendly API Design
+
+Terraform can work with any API, but some APIs are easier to deal with than others. Here are some tips to make your API more Terraform provider-friendly:
+
+### Think in terms of resources
+
+APIs that stick to a consistent set of arguments or parameters are much easier to map into Terraform. Terraform can describe the "thing" your API is managing. RPC-based APIs are harder to work with since they’re not declarative. Your HTTP APIs should describe REST-like actions (GET, POST, DELETE, etc.) against resources not remote procedures like: executeJob. 
+
+APIs designed around resources are easier to support than those focused on action-oriented endpoints.
+
+✅ Preferred: REST-like actions against resources
+
+```
+GET    /api/alerting/rule/{id}     # Kibana Alerting API
+POST   /api/alerting/rule/{id}     # Create rule
+PUT    /api/alerting/rule/{id}     # Update rule
+DELETE /api/alerting/rule/{id}     # Delete rule
+```
+
+⚠️ Alternative: action-style endpoints
+
+```
+POST /api/executeJob
+POST /api/processSpaceUpdate
+POST /api/triggerIndexRebalance
+```
+
+Mapping resource-oriented APIs to a declarative model means users can describe the desired state rather than the steps to achieve it. While action-oriented endpoints sometimes become necessary, knowing the trade-offs helps you make informed decisions.
+
+#### Ideal
+
+The [Index Lifecycle Management API](https://www.elastic.co/docs/api/doc/elasticsearch/operation/operation-ilm-put-lifecycle) (PUT /\_ilm/policy/{policy}) declaratively defines the desired lifecycle state rather than imperatively executing phase transitions:
+
+```terraform
+resource "elasticstack_elasticsearch_index_lifecycle" "my_ilm" {
+  name = "my_ilm_policy"
+
+  hot {
+    min_age = "1h"
+    rollover {
+      max_age = "1d"
+      max_size = "50g"
+    }
+  }
+
+  delete {
+    min_age = "60d"
+    delete {}
+  }
+}
+```
+
+[ILM resource](https://github.com/elastic/terraform-provider-elasticstack/blob/main/docs/resources/elasticsearch_index_lifecycle.md)
+
+#### API imposed challenges
+
+API design decisions can create real challenges for Terraform resource implementation. The Elasticsearch [Index Settings API](https://www.elastic.co/docs/api/doc/elasticsearch/operation/operation-indices-put-settings) (\`PUT /{index}/\_settings\`) treats static settings (which require index recreation) and dynamic settings (which can be updated in place) identically. This creates a complex situation forcing the implementation to handle all settings together because the API doesn't offer a way to distinguish them upfront. 
```

Comment:
```suggestion
API design decisions can create real challenges for Terraform resource implementation. The Elasticsearch [Index Settings API](https://www.elastic.co/docs/api/doc/elasticsearch/operation/operation-indices-put-settings) (\`PUT /\{index\}/\_settings\`) treats static settings (which require index recreation) and dynamic settings (which can be updated in place) identically. This creates a complex situation forcing the implementation to handle all settings together because the API doesn't offer a way to distinguish them upfront. 
```